anvil/observables.py

```diff
@@ -153,16 +148,24 @@ def bootstrap_function(func, states, *args, n_boot=100):
         dimension
 
     """
-    boot_index = torch.randint(0, states.shape[0], size=(states.shape[0], n_boot))
-    # put boot index on final dimension
-    resampled_states = states[boot_index, :].transpose(1, 2)
-    res = func(resampled_states, *args)
-    return res
+    # NOTE: Optimal seems to be around n_sample x batch_size = 10000 for
```

Comment:
surely this is also highly dependent on the lattice size?

Reply:
True. I wasn't sure it was worth bothering with though since for any reasonably large target length and `n_boot` (e.g. 10000, 1000) the fastest bootstrap "batch size" is one even for small lattices. Could be some efficiency gain but only for cases where the target length or `n_boot` is small which are fast anyway.

Reply:
but I will look into it actually

Reply:
I was doing some back of envelope stuff looking at tensor size and basically the objects created (assuming states are floats) are (N_bootbatch * N_lattice * target_length * 4 * 1e-9) gb which for 6x6 lattice, batch size 1000 and 10000 target length is like 1.44gb - not too crazy. With 8x8 this goes up to 2.56gb and then 10x10 is 4gb

something like the two point function means we have 2 copies of tensor like this so things can actually get pretty crazy pretty fast.